CMakeLists.txt

```diff
@@ -44,6 +44,18 @@ include(DefineOptions.cmake)
 include(GNUInstallDirs)
 include(FeatureSummary)
 
+set(DARKTABLE_LIBDIR    "${CMAKE_BINARY_DIR}/lib/darktable")
+set(DARKTABLE_BINDIR    "${CMAKE_BINARY_DIR}/bin")
+set(DARKTABLE_DATADIR   "${CMAKE_BINARY_DIR}/share/darktable") # darktablerc, themes, icons, ... Cannot use ${CMAKE_BINDIR}/data because not everything resides there. Like icons (pixmaps)
+set(DARKTABLE_SHAREDIR  "${DARKTABLE_DATADIR}")
+set(DARKTABLE_LOCALEDIR "${CMAKE_BINARY_DIR}/share/locale")
+
+set(DARKTABLE_RELATIVE_BINDIR    "./${CMAKE_INSTALL_BINDIR}")
+set(DARKTABLE_RELATIVE_LIBDIR    "../${CMAKE_INSTALL_LIBDIR}/darktable")
+set(DARKTABLE_RELATIVE_DATADIR   "../${CMAKE_INSTALL_DATAROOTDIR}/darktable")
```

Comment:
Is it even possible? Do you have any idea how to tackle that @parafin?

Reply:
I see 2 options:

1. add a check that CMAKE_INSTALL_BINDIR doesn't contain / or \ and fail with fatal error message if it does
2. generate correct relative paths by checking how deep is CMAKE_INSTALL_BINDIR (how many directory separators it contains)

Reply:
`DARKTABLE_RELATIVE_BINDIR` does not exist any longer and `CMAKE_INSTALL_BINDIR` is not used any more. It it still needed? If so, can you give me some hypothetical pathes that my PR should be able to work with?

Reply:
Erhm, what are relative paths relative to? You use CMAKE_INSTALL_BINDIR implicitly here. I already gave you a hypothetical path in first comment.

I'm fine with just erroring out. But until macOS case is tested/fixed it's not obvious if this approach going to work at all.

Reply:
I need a complete picture to rule out misunderstandings. Let me start an example based on your suggestion:
* /games/bin
* /lib/darktable/{libdarktable.so, plugins/, views}
* /share/{appdata, applications, darktable,  doc, locale, man}
So in this example using `..` would bring us to `/games`, not `/` and such the path resolution would fail and you suggest to count the directoriy levels from `CMAKE_INSTALL_BINDIR` to adjust for that - got it. So in the example above we would need `../..`.

Reply:
Yes, you are correct.

Reply:
I have recreated the path resolution using no hardcoded pathes. Everything is resolved from system specific cmake variables. I have left debug messages (turned off) in the code and will remove them when the code is tested. It should now cover all possible valid Posix use cases.